scrapers/StateOfTexas.mjs

```diff
@@ -5,27 +5,36 @@ export default ({ database, DataScraper }) => {
 	return DataScraper(database, 'StateOfTexas', async () => {
 		const data = [];
 		const baseUrl = 'https://api.tnris.org/api/v1';
-		const id = '2679b514-bb7b-409f-97f3-ee3879f34448';
-		const collections = `collections/${id}`;
-		const resources = `resources?collection_id=${id}`;
-		const collectionsRequest = await fetch(`${baseUrl}/${collections}`);
-		const collection = await collectionsRequest.json();
-		const resourcesRequest = await fetch(`${baseUrl}/${resources}`);
-		const resource = await resourcesRequest.json();
+		// let id = '2679b514-bb7b-409f-97f3-ee3879f34448';
+		const allcollections = await fetch(`${baseUrl}/collections_catalog`);
+		const result = await allcollections.json();
+		let collectionIds = result['results'];
+		// console.log(result['results'].length);
+		for(let i = 0;i<collectionIds.length;i++){
+			// console.log(collectionIds[i]['collection_id']);
+			let id = collectionIds[i]['collection_id'];
+			let collections = `collections/${id}`;
+			let resources = `resources?collection_id=${id}`;
+			let collectionsRequest = await fetch(`${baseUrl}/${collections}`);
+			let collection = await collectionsRequest.json();
+			let resourcesRequest = await fetch(`${baseUrl}/${resources}`);
+			let resource = await resourcesRequest.json();
 
-		const { publication_date, description, name } = collection;
-		const { results } = resource;
-		const created = moment(publication_date, 'YYYY-MM-DD').unix();
+			let { publication_date, description, name } = collection;
+			let { results } = resource;
+			let created = moment(publication_date, 'YYYY-MM-DD').unix();
 
-		results.map(item => {
-			return data.push({
-				url: item.resource,
-				updated: 0,
-				created,
-				description,
-				name,
+			results.map(item => {
```

Comment:
Although a problem with the original code, the usage of `map` is inappropriate and `forEach` should be used instead.